### ISSUE TITLE
Allow project administrators to conditionally show sidebar items

### ISSUE THREAD

**Issue by leolami** [g3w-suite/g3w-admin#1099]
### Checklist

- [x] I've searched through the [current issues](../issues) to make sure this feature hasn't been requested already.

### Motivation

When publishing a project, a form option should make it optional to display the session Metadata in the client

### Suggested solution

A simple check option in the form

**Comment by wlorenzetti** [comment]
@Raruto @volterra79 I suggest to implement it with more options, i.e.:

- Left-bar open on close at client runtime
- Set visibility of other sections: searches, bookmarks, print

How do you prefer this options inside the config ?

**Comment by volterra79** [comment]
@wlorenzetti i think that a configuration can be a project level: 

`sidebar: {
   print: { show: true},
   metadata: {show: false},
  ...
}`

@Raruto what dou you thin about?

**Comment by Raruto** [comment]
@wlorenzetti 

> Set visibility of other sections: searches, bookmarks, print

Generate some custom css (server side) and then include it on the page, eg:

```css
.sidebar-menu > li:is(#metadata, #spatialbookmarks, #print) { display: none; }
```
![image](https://github.com/user-attachments/assets/1c011128-af88-4955-803d-14152dd8ac2c)

> Left-bar open on close at client runtime

Maybe you missed: https://github.com/g3w-suite/g3w-admin/issues/944#issuecomment-2390639393

Ref: https://github.com/g3w-suite/g3w-client/blob/8c3e7a36eeb82af699b5889e87f10c1abec0ef01/src/index.html#L18

**Comment by wlorenzetti** [comment]
Yeas I can use a custom.css o custom.js, but we want add this as options available in the QGIS project uploading form:

![Screenshot_20241030_121958](https://github.com/user-attachments/assets/c66da55d-a23a-4bd6-b60d-ac2749c26acf)

**Comment by Raruto** [comment]
> Yeas I can use a custom.css o custom.js, but we want add this as options available in the QGIS project uploading form:

Yes, as I was saying, you just need to inject some css rules (on the fly) into client template:

```py
# somwhere within your djang app 

elements_to_hide = ['#metadata', '#spatialbookmarks', '#print'] # here your django options.. 😃
SETTINGS.CLIENT_CUSTOM_CSS = f".sidebar-menu > li:is({', '.join(elements_to_hide)})"
```

```py
# https://github.com/g3w-suite/g3w-client/blob/0322c7bf066eecd92617b7f4c4d7f06c2448ba77/src/index.html

{% if SETTINGS.CLIENT_CUSTOM_CSS %}
  <style>{{ SETTINGS.CLIENT_CUSTOM_CSS | safe }}</style>
{% endif %}
```





---

Then, if you're feeling fancy, you could even add two text areas to each project, so anyone can customize any project as they wish..

![image](https://github.com/user-attachments/assets/48bc24ca-3a88-4c5e-9266-97ad5fa456e0)

and then append it to any previous (server side generated) custom css:

```py
SETTINGS.CLIENT_CUSTOM_CSS += TEXT_AREA_CUSTOM_CSS
```
